physical/swift.go

```diff
@@ -63,12 +63,23 @@ func newSwiftBackend(conf map[string]string, logger log.Logger) (Backend, error)
 		tenant = conf["tenant"]
 	}
 
+	domain := os.Getenv("OS_USER_DOMAIN_NAME")
+	if domain == "" {
+		domain = conf["domain"]
+	}
+	tenantDomain := os.Getenv("OS_PROJECT_DOMAIN_NAME")
```

Comment:
This is a quirk that the OpenStack project introduced with KeyStone V3. Originally a KeyStone project was called a tenant and that naming was deprecated several releases ago. However, a lot of clients maintain the name "tenant" internally. I kept this name as the underlying `"github.com/ncw/swift` library still uses the old naming for projects. For the environment variable, I used the name that is required by the most recent versions of the OpenStack CLIs authenticating with KeyStone V3 API as this would present in most users' environments.